packages/dev-server-rollup/test/node/unit.test.ts

```diff
@@ -2,10 +2,47 @@ import { Plugin as RollupPlugin, AcornNode } from 'rollup';
 import { expect } from 'chai';
 import path from 'path';
 
-import { createTestServer, fetchText, expectIncludes } from './test-helpers';
+import { createTestServer, fetchText, expectIncludes, timeout } from './test-helpers';
 import { fromRollup } from '../../src/index';
 
 describe('@web/dev-server-rollup', () => {
+  describe('buildStart', () => {
+    it('awaits buildStart when starting a server', async () => {
+      let resolveBuildStartCalled: undefined | ((value: void) => void);
+      const buildStartCalledPromise = new Promise(resolve => (resolveBuildStartCalled = resolve));
+
+      let resolveBuildStart: undefined | ((value: void) => void);
+      const plugin: RollupPlugin = {
+        name: 'my-plugin',
+        buildStart() {
+          if (resolveBuildStartCalled) {
+            resolveBuildStartCalled();
+          }
+          return new Promise(resolve => (resolveBuildStart = resolve));
+        },
+      };
+
+      const serverPromise = createTestServer({
+        plugins: [fromRollup(() => plugin)()],
+      });
+
+      await buildStartCalledPromise;
+
+      try {
+        const res = await Promise.race([serverPromise, timeout(0, 'timeout')]);
+        expect(res).to.equal('timeout');
+        if (resolveBuildStart) {
+          resolveBuildStart();
+        }
+        const res2 = await Promise.race([serverPromise, timeout(0, 'timeout')]);
+        expect(res2).not.to.equal('timeout');
+      } finally {
+        const { server } = await serverPromise;
+        server.stop();
+      }
+    });
+  });
```

Comment:
this is pretty hard to test, I was a few time thinking maybe it's best to ditch this completely, also given it's disproportionate to the complexity of the logic and potential bugs in it

Reply:
I agree this is a pretty complex test just for sake of adding a missing `await`

Reply:
good, I'll remove it

Reply:
done